### DISCUSSION TITLE
Conflict - Do you fix problems quickly?- Chewing the Fat 📈

### DISCUSSION THREAD

**Discussion by piers-sinclair** [SSWConsulting/SSW.Rules.Content#5102] [Chewing the Fat]
Contributors: @UlyssesMaclaren @pierssinclairssw @adamcogan

Let's talk about this rule:
https://ssw.com.au/rules/fix-problems-quickly/

**Comment by GordonBeeming** [comment]
I like these kinds of topics. 

Software developers often don't focus on the people side of being a great developer. Later down, when some of the situations arise spoken about in some of the rules, like promotions or getting the "cool" project, don't go their way, they only realize the value of being able to communicate better with others in different roles and levels of experience, and to some degree have to play a lot of catchup.

**Comment by billyterblanche** [comment]
+1 for this weeks topic

I think communication is difficult to get right, but such an important skill to have. It affects our personal life and professional.

A good read related to this topic is Crucial Conversations: Tools for Talking When Stakes Are High

**Comment by premradhakrishnan** [comment]
I learnt about pushing back at the right time. Conflict management and resolution are skills every person needs to learn not just for at work but in everyday life too. Having a framework such as SSW Rules can certainly help with this.

**Comment by tiagov8** [comment]
```
.wife {
right: 100%;
margin: 0;
}
```

**Comment by 2fernandez** [comment]
.husband{
  float: right;
}

**Comment by AlexRothwell** [comment]
I think this is an important topic, especially as consultants; we need to manage our relationships with clients on a daily basis. Thankfully, so far at SSW all of my clients have been wonderful to work with, so conflict management and getting communications clear has not been an issue

**Comment by ozdemir-mehmet** [comment]
Always be respectful, even when you disagree. Avoid attacking the other person or their ideas and focus on the issue at hand.

**Comment by JackDevAU** [comment]
Communication is such a critical aspect in life, even more so working as a consultant. The rules have some good examples and tips to make sure we all understand how to avoid and fix problems :)

**Comment by apforsyth** [comment]
Conflict is part of life, it's only a question of how you approach it.  Sometimes I wish it wasn't!  I am not afraid of it, but I am not afraid of anchovies either (just disgusting!)

**Comment by Hona** [comment]
I think it’s all about how you frame the conflict. When approaching a problem, do you get defensive, do you reason it off? 

It’s always useful to remember the relationship rule of: you and me vs the problem, not you vs me - which is how these things can go

**Comment by matt-goldman** [comment]
There are two aspects to conflict management. The first is the one people talk about the most - managing emotions. This includes yours, and the other parties, or all parties if you're mediating/de-escalating.

The second, and perhaps most overlooked, is the actual issue. People get mad, there are misunderstandings; these things happen. But often, there is conflict because expectations have been broken. Avoiding conflict in the first place can be greatly helped by expectations management. This sounds like a fancy term, but it just means being honest and upfront. And once the expectations have been broken, you have to actually address the problem. Emotional management, conflict de-escalation, empathy - these are all great, but pointless if, amongst all that, you neglect to fix the problem that caused the conflict in the first place.

**Comment by antonvojkovicssw** [comment]
When it comes to dealing with conflicts, people should always approach the situations with the mindset of resolving the issue and not with the idea of things need to go the way I want or I think it's better.

**Comment by cvionnet** [comment]
The olive branch tip is a good one!
Trying to de-escalate conflicts as soon as possible, as it can rapidly have a big impact.

Resolving conflicts can be quite challenging when you're not trained and English is your second language, especially for choosing the most adaptable terms and vocabulary that avoid escalating things.

**Comment by samsmithssw** [comment]
Conflict happens in every aspect of life, the most important things to remember when involved are; to not act impulsively / out of emotion, and always make sure the other person feels validated and heard. I'm a firm believer that you can always reach a middle ground.

**Comment by Marxoz** [comment]
With so many personalities, cultures and opinions in the work place, conflict is inevitable. Everyone is entitled to their opinion and it is rare that a conflicting situation has a right and wrong person. That is why good communication is important in situations like this, with listening being the most important part. Always make an effort to listen and understand the other person's point of view rather than just forcing your point across. It will help with getting to a resolution quicker and peacefully.

Also, eat before you have difficult conversations. Being hangry does not help 😂

**Comment by AntPolkanov** [comment]
Nice idea to use emoji! GIFs might also be an option.

**Comment by piers-sinclair** [comment]
It's important to keep your eye on the prize, don't kick up a huge fuss if an issue doesn't really matter to you that much.

**Comment by chrishoogwerf** [comment]
We all communicate on different levels which sometimes gives out the wrong impressions. Seek to understand. 

Note to self:  Don't take everything personal, eat more food (as per gert).

**Comment by leehawkinsssw** [comment]
Creating an environment of psychological safety helps a lot in this regard. I see how the hyper-feedback culture in SSW easily leads to conflict, but Goldie's new rule around that is awesome (see https://www.ssw.com.au/rules/how-to-offer-unsolicited-feedback/). 

I encourage people to practice some kind of mindfulness/meditation to help with dealing with the emotions arising from situations of conflict. Being deliberate about taking pause between stimulus and response, especially in such stressful situations, can make a huge difference.

**Comment by leahy268** [comment]
As others have said.  Make sure you are talking about the actual problem not the emotions that come up.  
Hit it head on and soon after the issue happens so that you can get it resolved quickly, otherwise it just festers. 

I once did a Quality Control short course when I was still in school (read nerd).  The principal behind it was that the person is never the problem, it's always the process.  Personally I would say the person is 'rarely' the problem and 'usually' it's the process.  Sometimes it's the person though because they simply refuse to follow the process and don't want to change/don't want to work on the process.  😊

Also make sure that you have the conflict at least in video chat or even better in person.  Having a fight over email or text message is bad as it's easy to misinterpret written words as angry or upset.

**Comment by camillars** [comment]
Conflict at work can be a pain, but it's just part of any workplace. The best way to look at it is not to take it personally, as situations like these are normal, and then try to find the right way to approach the problem. Keep calm, deep breaths, understand where the other person is coming from and then reply with care. Sometimes all it takes is a small gesture of kindness and a smile to make things better! 🌷

**Comment by UlyssesMaclaren** [comment]
Be forward looking - don't focus overly on what happened... just enough to see there was a problem.

Instead, put most of your focus on what the 2 of you, together, can do to make the future better.

A large part of successful negotiation is about framing it as you 2 against the problem, rather than you against them (thank Chris Voss - Never Split the Difference - for that insight :) )

**Comment by amankumarrr** [comment]
Sometimes, our intentions are good when we try to push people to improve, but this approach may not work for everyone. Each person has their own unique way of learning, picking up things, work background, and values. Conflict can arise when we only consider our own frame of reference and fail to take into account the perspectives and needs of others.

**Comment by tombui99** [comment]
These are some great tips. Conflicts will always happen as it is the nature of human, it results in miscommunication and misunderstanding toward each other. However, the key is to always be respectful, never takes thing personally and never try to argue with anyone to win but try to ease the tension and provide a resolution. At the end of the day, everyone is trying to achieve the same goal.

**Comment by vladislav-kir** [comment]
This is an important topic. Conflicts happen in every aspects of life, would it be personal or work. It is important to focus on solving the issue together, rather then having an argument with a people with whom you do not agree.

**Comment by SamBWagner** [comment]
Humans are just that. Humans. Do not treat people with any elevated or reduced level of respect and you'll tend to do better. CEOs are just people as are the juniors who don't know what they're doing.

**Comment by chrisschultzssw** [comment]
So many good comments here; I'll just add:

📝 Using emoji is a good idea ✅💡, but use your judgement 🧑🏼‍⚖️ - they could come across as flippant 🙃.

**Comment by tkapa** [comment]
I learned these as "soft skills" but they're more important than any technical skill you can have. 

You can be the best in the world at your field, but no one will work with you if you're unpleasant to be around.

**Comment by Ben0189** [comment]
Resolving conflicts is important not only in the workplace but also in everyday life. Change in perspective is a other tip to solve conflict effectively

**Comment by jakebayliss** [comment]
This is a super important topic in the work place, its very dependent on both parties and their personalities. Different people handle conflict in different ways, some well and some not. I think the important thing is figuring out what works for you so that you can resolve things quickly

**Comment by bahjatalaadelssw** [comment]
Most conflict is an unresolved issue coming to the surface, and work tends to magnify anxiety due to stress. It's important to see if a root cause is fuelling the conflict.

Communication is key and timeliness will help with job satisfaction because employees will know that their concerns are being heard and addressed.

**Comment by zacharykeeping** [comment]
I like these tips. Conflict always seems to be inevitable but I think it's important to always maintain a calm and level head and work through these sorts of issues respectively.

**Comment by stephanfakossw** [comment]
Being assertive, keeping calm, patient and respectful. I found that it helps to park the issue for later when the confrontation levels are high.

**Comment by PennyWalker** [comment]
Communication is something you need to work on constantly. I think it's important that the person you're speaking to feels heard. It's ok to empathise with their position first, and then respectfully disagree. For example, "I hear you, I understand why you would want to choose Blazor. It's got some great features. Have you considered using Angular for the following reasons xxx"

In personal conflict, I find this still applies. Stay calm, cool, and collected, and try to avoid mirroring someone if they are upset or aggressive. Not mirroring their behavior can often de-escalate a situation. Sometimes it's better to continue a discussion later when people have a chance to cool down, and are not still digging a trench for their position. Timing is as important as delivery.

**Comment by AshAnil** [comment]
This week's topic is so important. Conflicts are common, but the way we deal with them can make all the difference in the outcome. 
Those tips from the rule are the best solution to avoid conflicts. #Staycalm #Patience #communication

**Comment by jimmy1992abc** [comment]
I like this topic more than any others. This topic is important because emotion plays a big part in our ordinary work, conflict and argument are always happening. People enjoy their work not only considering their remuneration, but also a comfortable and lovely environment. 

I love the olive rule and the time when to push back. I believe the best solution is always an appropriate way to release the emotions, not ignore or freeze them.

In addition, I believe a good third party is always a good choice.

**Comment by Harry-Ross** [comment]
I thought this was a really useful topic, providing a variety of helpful information about resolving conflict in the workplace. It is always important to ensure that this is at the forefront of our minds when dealing with potential issues you may encounter with colleagues and clients. I thought this is a really helpful refresher to ensure that everyone is following appropriate etiquette when solving workplace conflicts.

**Comment by KostaMadorsky** [comment]
Proper conflict management is important when dealing with clients and colleagues.

**Comment by steven0x51** [comment]
I like this week's topic, these soft skills are useful, especially when dealing with clients and colleagues

**Comment by wicksipedia** [comment]
I have to deal with this far too much at home. Miss 4 & expectation management is difficult as she's experiencing lots of emotions (especially when tired) and it's difficult for her to manage/navigate them. Most of the time, escalation is inevitable. Eventually we find out (most of the time) she doesn't feel like she's being heard.

Whilst this isn't a work scenario, I feel it's still applicable to work scenarios.

I think acknowledging someone's problem, discussing a solution, and giving them a timeframe goes a long way to diffusing a situation

**Comment by drwharris** [comment]
I'm a big fan of "Make client complaints a positive experience"

I think everyone appreciates when a complaint is taken serious and a positive outcome for both can be achieved.

**Comment by AdriT9** [comment]
Learning & knowing how to deal with conflict in a healthy & positive way, is essential, as conflict can be disruptive and damaging. 

It's important to mention that dealing with conflict in a healthy and positive way can be challenging because of the intensity of emotions, communication difficulties, different perspectives, power dynamics, and past experiences. Approaching conflict resolution with an open mind, a willingness to listen, and a commitment to finding a resolution that works for everyone involved (all easier said than done!) 🥲

**Comment by christianmorfordwaitessw** [comment]
It's good for people to resolve the conflict by understanding both sides so that it can be avoided in the future

**Comment by JackDuan123** [comment]
How to deal with confilict?
Just control your anger

**Comment by Levijj22** [comment]
Some people love confrontation, and some even look for it. 
You can usually easily identify these people. Sometimes it’s best not to engage in heated conversations with these types of people as the issue might not even be a real issue!

**Reply by JackDuan123** [reply]
I love confrontations on court ,
Confrontation is a strategy to win sometimes.
And if you want to win, it is always having a confrontation and control the madness

**Comment by landonmaxwell** [comment]
I think conflict often follows the same stages as grief: denial, anger, bargaining, depression, acceptance.
Fast forward the process straight to acceptance and save yourself a lot of grief 😂

**Comment by Abf09** [comment]
I think communication is key to resolving any conflict, listening and acknowledging the issue solves most of the conflict, making it a win-win for both parties to fix it completely

**Comment by KahaMason** [comment]
Conflicts in personalities are always a clear and present possibility. Understanding the other person's intent is critical. You can't always control how others behave / react but it's always a good idea to remain calm and respectful but be strong and firm when focusing / dealing with the issue at hand.

**Reply by KahaMason** [reply]
Building on my comment.

Communication is crucial. Conflicts may not even need to arise if you take a more proactive approach with communicating. Understanding the other person's concerns and being receptive and reassuring that they have been heard, is a great way of resolving issues before they escalate. A simple conversation or a call is better than no communication at all.

If you get an unexpected emoij reaction, it is best to double check by calling.

Ignoring a problem is never a good way to go.

**Comment by christoment** [comment]
Overall this is a good rule!

I have a couple of comments though:

> One-off task - Most of the time, even if you're unhappy with some work you've been given, but it's a one-off task, don't stress the small stuff, and just get it done

I think we still have to send a record that this is something that is not pleasant to do or not wanting to do.

> Avoid bad blood

I think emojis shouldn't be use in some cases since it shows a playful tone instead of a apologetic one.

That's my 2cents :)

**Reply by samsmithssw** [reply]
I agree, I think the olive branch emoji comes across as insincere and 'a forced' apology

**Comment by hveraus** [comment]
Give the other person a chance to tell his or her side of the conflict. Relax and listen; try to learn how the other person feels.

**Comment by jaydenalchin** [comment]
Overall a great rule and any content about conflict resolution carries a lot of inherent value. 

My thoughts and feedback: 
- Seems like this rule reiterates a lot of existing rules content
  - e.g. https://www.ssw.com.au/rules/how-to-take-feedback-or-criticism
  - e.g. https://www.ssw.com.au/rules/follow-up-effectively/
  - The rule could benefit from more clarity of purpose
- The emojis are fun but used in some inappropriate less-fun cases here (olive branch, talking about bad blood)
- This and some related rules carry unnecessary prefixes in their title 😵‍💫😪

**Comment by jerryluossw** [comment]
It's perfectly normal to have conflicts at work, for me, the following rules are important:
1. Listen more to what he means and try not to misunderstand
2. Be aware that the purpose of the conflict is to solve the problem, not to get personal

**Comment by SofieHong** [comment]
Instead of rushing to dismiss other people's ideas, try to analyze the pros and cons of different ideas and solutions together.  🤝
Positive communication can resolve most conflicts. ☕

**Comment by AttackOnMorty** [comment]
What if some people are still unreasonable after you've tried everything? 
Ignore them if possible. You can't convince everyone.

**Comment by KristenHu** [comment]
Throwing out an olive branch is a good way to avoid unresolved conflict, and it is also important for us to choose our words carefully when we have conflicts with others.

**Comment by sylhuang** [comment]
Miscommunication and conflict are not always equivalent, although the former usually leads to the latter. From my experience, 'proving you're right' is not the key to resolve the conflict. Making people happy and willing to step back for each other and embrace the difference would work most of time (yet even harder to achieve than proving you're right)

**Comment by AnkyChe** [comment]
Love this week's topic. Conflict is inevitable so it's important to learn how to deal with it. Bad moods may escalate the conflicts. So I think calm down and put yourself in someone else's shoes is important.

**Comment by Berber-Mama** [comment]
Looking after the moods during communication is always a good tip. Focus on the issue, not the person.

**Comment by jimmidier** [comment]
Conflicts reflect differenct stance and interest of people, and the way to address them generally depends on the specific scenario, as in whether it's a centralized system where someone owns the final decision, or a decentralized system where there's no trusted authority. Sacrifice is essentially inevitable though.

**Comment by johnxu16** [comment]
It is always difficult to resolve conflicts, especially encountering inevitable competition of interests.
There are a lot of good tips and they help me to learn how to win non-constant-sum games.

**Comment by calumjs** [comment]
Conflict in the workplace can have advantages when managed effectively. It can lead to increased creativity and innovation, improved decision-making, increased motivation and engagement, stronger relationships, and personal growth and development. However, it's important to approach conflict with care and to work towards finding positive and productive resolutions. Conflict that is left unresolved or that escalates can have serious negative consequences for individuals and for the organization as a whole.

**Comment by Freego1783** [comment]
Really nice topic, communication is so important.

Stay calm and don't get defensive ! People tend to take everything personally in these situations.

**Comment by rubycogan** [comment]
Great topic. SSW focuses a lot on avoiding misscomunations, and this is an excellent addition to it. More specific tips on how to manage your emotional response first would be good as dealing with the problem properly is much harder when you still feel very upset.

**Comment by BrookJeynes** [comment]
Communication is such an important thing to have in the workplace. Giving both sides a chance to tell their story and coming into a conflict with the mindset of growth are two ways we can help resolve conflicts

**Comment by UlyssesMaclaren** [comment]
Did Chat GPT write this? 🤣😂











Ulysses Maclaren   |   SSW General Manager<http://www.ssw.com.au/People/Ulysses-Maclaren?utm_source=emailsignature>   |    +61 419 418 232<https://www.ssw.com.au/ssw/Company/Offices/?utm_source=emailsignature>   |   Ping me on ***@***.***>

**Reply by calumjs** [reply]
You got me

**Comment by danielmackay** [comment]
This is a valuable topic, and through being aware of how certain people might feel in some situations we can avoid conflicts and create a much more pleasant workplace.

I loved the rules about [fairness](https://www.ssw.com.au/rules/speak-up/) and understanding decisions you [may not agree with](https://www.ssw.com.au/rules/lose-battle-keep-client/#how-to-understand-and-rationalize-dissenting-opinions).

On a similar topic I also think it helps to have the right mindset when receiving feedback.  Sometimes there may be a tendency to take the feedback as negative and get defensive.  In reality, the majority of feedback comes from a good place and a desire for you to improve.

**Comment by TobyChurches** [comment]
This is definitely a great area to discuss as part of Chewing the Fat! It is only natural to have conflicting ideas in the workplace, however, it is essential that these disagreements are dealt with appropriately. Even if you disagree with an idea it's important that you take the time to try and understand their point without completely dismissing it.

**Comment by jontygardner** [comment]
This is a great topic! 👏🏽 

The 3 biggest tips I love from this are:

1. Stay calm and listen. 

2. Action the issue asap 

3. Be transparent and brainstorm solutions together

Address the issue as an external thing and not a personal attack. Don't let emotions get in the way of tackling an issue and resolving something that is bugging someone. E.g. If your scum master is getting annoyed at you for sending your daily scrum emails late, don't be personally offended, take it as a growing moment for your professional career. Set yourself an automated reminder every day at 9:15 AM and tell your scrum master you have done this so you should be much better at sending it out on time!

**Comment by jernejk** [comment]
I believe most hard to manage conflicts on projects come from a different source then it appears.

For instance, I had some non-SSW developers openly criticising on how we work and our code, only to come to understanding they are either threaten for their job/pay or that they’ll have to continue working on the project and they would get the blame if things are not working after we left. I usually resolve this behaviour by understanding not only the things they complain about but also how would they have done and if they believe that for given info and time, they would have done a better job. More often then not, we end up on friendly terms but it can take weeks or months before they understand what they are doing isn’t helping anyone.

I had Product Owners who were stressed about delivery dates all the time even though we were on time and we had conflicts whether or not we’ll deliver. It turns out, they had many developers in the past who massively underestimated and even hid the fact they were very late. In many cases, these things are not subconscious and you find out when you find patterns in behaviour or how various suggested solutions make them feel. When understood that, I had 1:1 chat with the person every 2-days for 5 minutes on top of daily scrum and brief them on how I feel we are going with the time and what he can do to make sure we have no blockers. It made no change to the project but it made the client feel more at ease. (Of course we delivered 😊)


In short, I like to understand the nature of the conflict as that allows me to adjust my angle and whether I need to involve more people to resolve the conflict like Uly/Penny or Adam. As long the conflict isn’t malicious or political, they can be resolved through understanding and action. Many conflict may happen out of logical but many escalates because we are human.

I think it’s important to understand that in many cases we enter in the projects as outsiders.

Some client’s developers may not look at us as people who are there to help them but people who threaten their positions or even jobs. They might not know why they weren’t good enough for the project or once we leave, will they be on their own dealing with everything and get blamed for our work. They might unintentionally cause conflicts to solidify their importance.

And for Product Owners, we are liability for them from day one, as they don’t know us and they might be responsible if we fail. It can be a heavy toll on them when things are not going well. They might cause conflicts out of stress and the fear of unknown.

**Comment by Rathi96** [comment]
Looking to enhance your product delivery process? Check out this resource on managing spilled stories for better predictability here. It’s a game-changer! https://www.kaiburr.com/blog/how-to-improve-predictability-of-product-delivery-by-managing-spilled-stories/

**Comment by yaqi-lyu** [comment]
For every task, we should respond as soon as possible even though things can't be fixed immediately. Keeping people updated prevents confusion and shows that we care and am taking it seriously.
